cpp/recognition/planar/HopcroftTarjan.cpp

```diff
@@ -0,0 +1,358 @@
+                dfs(g, u, reached);
+            }
+        }
+    }
+
+    void HopcroftTarjan::dfs_in_make_biconnected_graph(
+            NetworKit::Graph &g, NetworKit::node v, int &dfs_count,
+            std::vector<bool> &reached, std::vector<NetworKit::node> &dfsnum,
+            std::vector<NetworKit::node> &lowpt, std::vector<NetworKit::node> &parent) {
+        NetworKit::node w = NetworKit::none;
+        dfsnum[v] = dfs_count++;
+        lowpt[v] = dfsnum[v];
+        reached[v] = true;
+        std::vector<int> neighbors;
+        for (auto u : g.neighborRange(v)) {
+            neighbors.push_back(u);
+        }
+        for (auto u : neighbors) {
+            if (!reached[u]) {
+                if (w == NetworKit::none) {
+                    w = u;
+                }
+                parent[u] = v;
+                dfs_in_make_biconnected_graph(g, u, dfs_count, reached, dfsnum,
+                                              lowpt, parent);
+                if (lowpt[u] == dfsnum[v]) {
+                    if (u == w && parent[v] != NetworKit::none) {
+                        g.addEdge(u, parent[v]);
+                        g.addEdge(parent[v], u);
+                    }
+                    if (u != w) {
+                        g.addEdge(u, w);
+                        g.addEdge(w, u);
+                    }
+                }
+                lowpt[v] = std::min(lowpt[v], lowpt[u]);
+            } else {
+                lowpt[v] = std::min(lowpt[v], dfsnum[u]);
+            }
+        }
+    }
+
+    void HopcroftTarjan::make_biconnected_graph(NetworKit::Graph &g) {
+        // make it connected
+        NetworKit::node u = 0;
+        std::vector<bool> reached(N);
+        for (auto v : g.nodeRange()) {
+            if (!reached[v]) {
+                dfs(g, v, reached);
+                if (u != v) {
+                    g.addEdge(u, v);
+                    g.addEdge(v, u);
+                }
+            }
+        }
+        // make it biconnected
+        fill(reached.begin(), reached.end(), false);
+        std::vector<NetworKit::node> lowpt(N);
+        std::vector<NetworKit::node> parent(N, NetworKit::none);
+        std::vector<NetworKit::node> dfsnum(N);
+        int dfs_count = 0;
+
+        dfs_in_make_biconnected_graph(g, u, dfs_count, reached, dfsnum, lowpt,
+                                      parent);
+    }
+
+    void HopcroftTarjan::dfs_in_reorder(NetworKit::Graph &g, NetworKit::node v,
+                                        std::vector<NetworKit::Edge> &Add,
+                                        int &dfs_count, std::vector<bool> &reached,
+                                        std::vector<NetworKit::node> &dfsnum,
+                                        std::vector<NetworKit::node> &lowpt,
+                                        std::vector<NetworKit::node> &lowpt2,
+                                        std::vector<NetworKit::node> &parent) {
+        dfsnum[v] = dfs_count++;
```

Comment:
Again, it is advisable to go with DFSFrom method from NetworKit library - unless you have a really good reason not to do so.